tomcat/src/main/java/org/apache/catalina/SessionManager.java

```diff
@@ -0,0 +1,24 @@
+package org.apache.catalina;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class SessionManager implements Manager {
+
+    private static final Map<String, Session> SESSIONS = new HashMap<>();
+
```

Comment:
```suggestion
 private static final SessionManager INSTANCE = new SessionManager();
    private final Map<String, Session> sessions = new HashMap<>();

    private SessionManager() {}

    public static SessionManager getInstance() {
        return INSTANCE;
    }

```

SessionManager는 애플리케이션이 동작할 때 반드시 1개만 존재해야하니까 아예 클래스를 싱글톤으로 만들어버리는 건 어떻게 생각하시나요~?

지금도 Http11Processor에서 private static final로 주입해주고있어서 문제는 없겠으나, 생성자는 현재 public이어서, 추후에 여러개의 SessionManager Instatnce가 존재할 수도 있을 것 같아요!

Reply:
좋습니다! SessionManager가 중복해서 생성되는 경우를 막을 수 있겠군요 👍

[refactor: SessionManager 싱글톤 패턴 적용](https://github.com/woowacourse/java-http/pull/905/commits/4f29ad35c7ede20687892962d4b55fecd74700ed)